amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -115,9 +134,28 @@ private void cleanTestResources(StreamExistenceManager streamExistenceManager, L
     }
 
     private void startProducer() {
-        // Send dummy data to stream
-        this.producerExecutor = Executors.newSingleThreadScheduledExecutor();
-        this.producerFuture = producerExecutor.scheduleAtFixedRate(this::publishRecord, 60, 1, TimeUnit.SECONDS);
+        producerExecutor = Executors.newSingleThreadScheduledExecutor();
+        producerFuture = producerExecutor.scheduleAtFixedRate(this::publishRecord, 10, 1, TimeUnit.SECONDS);
+
+        // Reshard logic if required for the test
+        if (consumerConfig.getReshardConfig() != null) {
+            log.info("------------------------- Reshard Config found -----------------------------");
+            final Queue<ReshardOptions> reshardQueue = new LinkedList<>(Arrays.asList(consumerConfig.getReshardConfig().getReshardingFactorCycle()));
+            int totalRotations = reshardQueue.size() * (consumerConfig.getReshardConfig().getNumReshardCycles() - 1);
+
+            final StreamScaler s = new StreamScaler(kinesisClient, consumerConfig.getStreamName(), reshardQueue, totalRotations, consumerConfig);
+
+            Runnable task1 = () -> {
+                log.info("----------------------Starting new reshard------------------------------");
+                s.run();
+            };
+
+            // Split shard
+            producerExecutor.schedule(task1, 2, TimeUnit.MINUTES);
+
+            // Merge shard
+            producerExecutor.schedule(task1, 6, TimeUnit.MINUTES);
```

Comment:
Does this assume the contents of the reshard config? What if a future config is merge-split, or split-merge-split?